src/version.c

```diff
@@ -161,6 +162,17 @@ int check_versions(int *current_version,
 		return -1;
 	}
 
+	if (requested_version != -1) {
+		if (requested_version <= *current_version) {
+			fprintf(stderr, "Requested version for update (%d) must be greater than current version (%d)\n",
+				requested_version, *current_version);
+			return -1;
+		}
+		if (requested_version < *server_version) {
+			*server_version = requested_version;
+		}
+	}
+
 	//TODO allow policy layer to send us to intermediate version?
```

Comment:
This isn't my full review yet, but looks like this TODO isn't needed anymore, since this PR adds a way to get to an intermediate version.

Reply:
Removed.